mapping-for-navigation/_posts/0000-01-11-dual-carriageways.md

```diff
@@ -0,0 +1,119 @@
+based on the source of data, local community best practices and personal preference. This guide distills these best
+practices on how to improve road network data in OpenStreetMap by re-mapping bi-directional highways to dual-carriageways.
+
+Our basic rule of thumb are:
+
+*   Keep things simple, add complexity only when necessary.
+*   Dual-carriageway should have a physical barrier visible from imagery or in some cases road marking that show its illegal to pass on the other side (MUTCD).
+*   There is enough length from the physical divider (~20 meters?).
+*   During remapping, all other elements associated to the road should be re-tagged (for example, turn lanes, turn restrictions, route relation, etc.)
+
+## Changing from a simple bi-direction to a dual-carriageway
+
+<!-- https://www.openstreetmap.org/#map=18/37.955426/-121.281075 -->
+
+|Before|After|
+|---|---|
+|[![][4ae13b0b]][4ae13b0b]|[![][082fb704]][082fb704]|
+
+  [4ae13b0b]: {{site.baseurl}}/images/dual-carriageway/add-a-way-before.png "Click to enlarge"
+  [082fb704]: {{site.baseurl}}/images/dual-carriageway/add-a-way-after.png "Click to enlarge"
+
+**Option 1**
+
+1.  Select the bi-directional road and check if there are no special tags (such as turn-lanes, route relations, etc)
+2.  Align the original highway to one side of the dual-carriageway.
+3.  Trace the other segment of the dual-carriageway.
+4.  Copy the tags from the original highway to the newly created way.
+5.  Add **oneway=yes** for both ways.
+6.  Check for alignments and, etc. ![]({{site.baseurl}}/images/dual-carriageway/simple-add-a-way.gif)
+
+**Option 2**
+
+1.  Align the original highway to one side of the dual-carriageway.
+2.  Select the original way.
+3.  Click the  **Make parallel copies of ways** (<kbd>Shift</kbd>+<kbd>P</kbd>) then drag to create a new way parallel
+to the selected way.  This copies all the tags from the reference way.
+4.  Connect and align both ways to the road network. Use **Merge Nodes** (<kbd>M</kbd>) or **Join Node to Way** (<kbd>J</kbd>).
+5.  Add **oneway=yes** to both ways.
+6.  Reverse (<kbd>R</kbd>)way direction of the copied way. ![]({{site.baseurl}}/images/dual-carriageway/parallel-a-way.gif)
+
+## Fixing turn lanes
+
+<!-- https://www.openstreetmap.org/#map=20/37.83716908954435/-122.30457356404747 -->
+
+|Before|After|
+|---|---|
+|[![][a907d766]][a907d766]|[![][55088a6b]][55088a6b]|
+
+  [a907d766]: {{site.baseurl}}/images/dual-carriageway/turnlanes-before.png "Click to enlarge"
+  [55088a6b]: {{site.baseurl}}/images/dual-carriageway/turnlanes-after.png "Click to enlarge"
+
+We need to use the [Lanes and road attributes](https://josm.openstreetmap.de/wiki/Styles) map style and the [Turnlanes tagging](https://github.com/JOSM/turnlanes-tagging) plugin in JOSM. Check this guide on how to [activate the style and plugin](http://127.0.0.1:4000/mapping/mapping-for-navigation/adding-turn-lanes/#pre-requirements-for-mapping-a-turn-lane-in-josm) in JOSM.
```

Comment:
localhost link should be converted into the internal link

Reply:
fixed